packages/gradient-block/src/components/SquareBadge.tsx

```diff
@@ -50,14 +53,22 @@ export const SquareBadge = ({ gradientColor, gradientOrientation, index, blockWi
         }
     };
 
+    useEffect(() => {
+        if (badgeRef.current && badgeRef.current.clientWidth + badgeRef.current.offsetLeft > blockWidth) {
+            setOutOfBounds(true);
+        }
+    }, [badgeRef, blockWidth]);
```

Comment:
We can remove this useEffect and the outOfBounds useState, and just have a variable.

`const isOutOfBounds = !!badgeRef.current && badgeRef.current.clientWidth + badgeRef.current.offsetLeft > blockWidth`